src/lib/operators/table_scan/like_table_scan_impl.cpp

```diff
@@ -131,6 +90,166 @@ void LikeTableScanImpl::handle_column(const BaseDictionaryColumn& base_column,
+    });
+
+  } else if (pattern_variant.type() == typeid(EndsWithPattern)) {
+    const auto& suffix = boost::get<EndsWithPattern>(pattern_variant).string;
+    functor([&](const std::string& string) -> bool {
+      if (string.size() < suffix.size()) return invert_results;
+      return (string.compare(string.size() - suffix.size(), suffix.size(), suffix) == 0) ^ invert_results;
+    });
+
+  } else if (pattern_variant.type() == typeid(ContainsPattern)) {
+    const auto& contains_str = boost::get<ContainsPattern>(pattern_variant).string;
+    functor([&](const std::string& string) -> bool {
+      return (string.find(contains_str) != std::string::npos) ^ invert_results;
+    });
+
+  } else if (pattern_variant.type() == typeid(MultipleContainsPattern)) {
+    const auto& contains_strs = boost::get<MultipleContainsPattern>(pattern_variant).strings;
+
+    functor([&](const std::string& string) -> bool {
+      auto current_position = size_t{0};
+      for (const auto& contains_str : contains_strs) {
+        current_position = string.find(contains_str, current_position);
+        if (current_position == std::string::npos) return invert_results;
+      }
+      return !invert_results;
+    });
+
+  } else if (pattern_variant.type() == typeid(std::regex)) {
+    const auto& regex = boost::get<std::regex>(pattern_variant);
+
+    functor([&](const std::string& string) -> bool { return std::regex_match(string, regex) ^ invert_results; });
+
+  } else {
+    Fail("Pattern not implemented. Probably a bug.");
+  }
+}
+
+std::string LikeTableScanImpl::sql_like_to_regex(std::string sql_like) {
+  // Do substitution of <backslash> with <backslash><backslash> FIRST, because otherwise it will also replace
+  // backslashes introduced by the other substitutions
+  constexpr auto replace_by = std::array<std::pair<const char*, const char*>, 15u>{{{"\\", "\\\\"},
+                                                                                    {".", "\\."},
+                                                                                    {"^", "\\^"},
+                                                                                    {"$", "\\$"},
+                                                                                    {"+", "\\+"},
+                                                                                    {"?", "\\?"},
+                                                                                    {"(", "\\("},
+                                                                                    {")", "\\)"},
+                                                                                    {"{", "\\{"},
+                                                                                    {"}", "\\}"},
+                                                                                    {"|", "\\|"},
+                                                                                    {".", "\\."},
```

Comment:
this line is a duplicate of [line 221](https://github.com/hyrise/hyrise/pull/858/files#diff-e28a58ac7edaf897697f158c26da1b95R221)

Reply:
htf did you even spot that.